editor/editor_build_profile.cpp

```diff
@@ -567,6 +568,12 @@ void EditorBuildProfileManager::_detect_classes() {
 	}
 }
 
+void EditorBuildProfileManager::_detect_build_options() {
+	ProjectSettings *project_settings = ProjectSettings::get_singleton();
+	bool xr_disabled = !project_settings->get_setting_with_override("xr/openxr/enabled");
+	edited->set_disable_build_option(EditorBuildProfile::BuildOption::BUILD_OPTION_XR, xr_disabled);
```

Comment:
IIUC it should check it by itself in ` EditorBuildProfile::save_to_file`...

https://github.com/godotengine/godot/blob/01152ab664a4743e0076e9ba91c2436ae577550c/editor/editor_build_profile.cpp#L242-L250

Reply:
>umh, that's really weird, but I don't know this part of the engine well, so I'm not sure I can help further without studying the whole thing.

That part of the engine is whole weird because of several other bugs https://github.com/godotengine/godot/issues/89478 and https://github.com/godotengine/godot/issues/89477

It seems to me that I found on place that sets the value in another way and I will check it today or yesterday...

Reply:
About alternative way to set value... I just messed up and it was another thing but with name very similar to method that can set it is value

And it looks like it isn't magic regression because I've just checked 4.2.1-stable and I have the same problem although I thought it wasn't there 😅

The problem was in fact that it creates anyway creates dictionary which is then serialized in json even if it is empty so I just needed to add the checks `if empty do nothing`